torch/distributed/distributed_c10d.py

```diff
@@ -775,6 +779,21 @@ def get_backend_config(group: Optional[ProcessGroup] = None) -> str:
     assert backend_config is not None
     return str(backend_config)
 
+def _get_device(group: Optional[ProcessGroup] = None) -> torch.device:
+    """
+    Returns the device of the given backend.
+    If one ProcessGroup has multiple devices, the earlier the device is, the higher the priority.
+    For example, while user set ProcessGroup's backend as 'cpu:gloo,cuda:gloo',
+    'cpu' has higher priority and will be returned by thi func.
+    """
+    backend = get_backend(group)
+    priority_backend = backend.split(",")[0]
+    device = priority_backend.split(":")[0]
+    device_module = getattr(torch, device, None)
+    if device_module is None or device == "cpu":
```

Comment:
Why `cpu` is special cased here? `torch.cpu` does exist

Reply:
because `torch.cpu` does not have `current_device`. I think it's ok to execute original code when device is `cpu`.

Reply:
Can we just add a `current_device` to torch.cpu? :D